include/ccf/historical_queries_interface.h

```diff
@@ -7,17 +7,13 @@
 #include "ccf/receipt.h"
 #include "ccf/seq_no_collection.h"
 #include "ccf/tx_id.h"
+#include "ccf/tx_receipt.h"
 
 #include <chrono>
 #include <memory>
 
 namespace ccf
 {
-  // This is an opaque, incomplete type, but can be summarised to a
-  // JSON-serialisable form by the functions below
-  struct TxReceipt;
-  using TxReceiptPtr = std::shared_ptr<TxReceipt>;
-
   ccf::Receipt describe_receipt(
     const TxReceipt& receipt, bool include_root = false);
```

Comment:
Naming is hard, as this function signature makes clear. I think we should rename either `ccf::Receipt` or `ccf::TxReceipt`. The former is harder to rename because it's already public. Maybe the latter should be `BinaryReceipt`, or `RawReceipt`, or something, to highlight it contains directly useable hash byte arrays, rather than a hex/b64 encoding?

Reply:
I think `ccf::Receipt` should be the binary type name for a C++ API, but since it's been public already that's hard to do :-/ SerializedReceipt, B64Receipt, StringReceipt?

Reply:
I've gone back and forth on this, and I'm going to rename the current `ccf::Receipt` type. It seems clear that the old names are the wrong way round, and I'd rather bite the bullet early rather than later.

Reply:
+1 that's what I would do too, I hope it won't trigger too much confusion!